java/client/src/org/openqa/selenium/remote/internal/HttpClientFactory.java

```diff
@@ -114,18 +119,18 @@ public HttpClient getGridHttpClient(int connection_timeout, int socket_timeout)
         .build();
   }
 
-  private SocketConfig createSocketConfig() {
+  private SocketConfig createSocketConfig(int socketTimeout) {
     return SocketConfig.custom()
         .setSoReuseAddress(true)
-        .setSoTimeout(TIMEOUT_THREE_HOURS)
+        .setSoTimeout(socketTimeout > 0 ? socketTimeout : TIMEOUT_THREE_HOURS)
```

Comment:
The validation looks good.  But I'm still a bit confused on the `socketTimeout > 0 ? socketTimeout : TIMEOUT_THREE_HOURS` check.  I thought you indicated than was necessary for validation.